strangler-infusereo/code/ruby-service.rb

```diff
@@ -0,0 +1,37 @@
+require 'sinatra'
+require 'json'
+require 'net/http'
+
+set :port, 9494
+
+$users=[{name: 'foo', id: 1}]
+
+$delegate_user_call=false
+
+get "/v1/user" do
+  content_type :json
+  $users.to_json
+end
+
+get "/v1/user/:id" do
+  content_type :json
+  if ($delegate_user_call)
+    # call golang endpoint and return it
+    url = URI.parse('http://localhost:8080/v1/users/' + params[:id].to_s)
+    req = Net::HTTP::Get.new(url.to_s)
+    res = Net::HTTP.start(url.host, url.port) {|http|
+      http.request(req)
+    }
+    return res.body # already in JSON
+  end
+
+  user = get_user_by_id(params[:id].to_i)
+  if user != nil
```

Comment:
This section would likely be better expressed either this way:

return (user.nil? ? 404 : user.to_json)

or this way:

ret_val = 404
ret_val = user.to_json if !user.nil?

It avoids two returns and may be a little cleaner? Ternaries are ugly, but that makes it one line and eliminates the need for another variable.